### PR TITLE
Security: Add new setting allow_embedding

### LINKED ISSUE #14189: Clickjacking security issue: Missing X-Frame-Options header
Grafana does not set the `X-Frame-Options` header, which makes it vulnerable to [clickjacking](https://www.owasp.org/index.php/Clickjacking).

We run a bug hunting program, and got reported this issue. Clickjacking is considered not as bad practice, but as security issue, see the same issue on [yelp](https://hackerone.com/reports/49888) or [mail.ru](https://hackerone.com/reports/261652)

Affected Grafana versions: all, see https://securityheaders.com/?q=https%3A%2F%2Fplay.grafana.org.

### Vulnerability Description

Clickjacking (User Interface redress attack, UI redress attack, UI redressing) is a malicious technique of tricking a Web user into clicking on something different from what the user perceives they are clicking on, thus potentially revealing confidential information or taking control of their computer while clicking on seemingly innocuous web pages.

The server didn't return an X-Frame-Options header which means that this website could be at risk of a clickjacking attack. The X-Frame-Options HTTP response header can be used to indicate whether or not a browser should be allowed to render a page in a <frame> or <iframe>. Sites can use this to avoid clickjacking attacks, by ensuring that their content is not embedded into other sites.
This vulnerability affects Web Server.

### Steps to reproduce the vulnerability
Create a simple html file with the following content and open it in a browser:
```
<html>
 <head>
  <title>Clickjack test page</title>
 </head>
 <body>
  <p>Website is vulnerable to clickjacking!</p>
  <iframe src="https://play.grafana.org" width="1500" height="1000"></iframe>
 </body>
</html>

```

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
When allow_embedding is false (default) the Grafana backend will set the http header
`X-Frame-Options: deny` in all responses to non-static content which will instruct 
browser to not allow Grafana to be embedded in `<frame>`, `<iframe>`, `<embed>` or `<object>`.

**Which issue(s) this PR fixes**:
Closes #14189

**Special notes for your reviewer**:

### CODE REVIEW

File: conf/defaults.ini [hunk 1]
```diff
@@ -176,6 +176,9 @@ cookie_secure = false
 # set cookie SameSite attribute. defaults to `lax`. can be set to "lax", "strict" and "none"
 cookie_samesite = lax
 
+# set to true if you want to allow browsers to render Grafana in a <frame>, <iframe>, <embed> or <object>. default is false.
+allow_embedding = false
```
Comment (bergquist) [inline conf/defaults.ini]: :+1: We should default the secure option as long as it's still easy to get started on a local dev machine.

**Review by bergquist — APPROVED** [review]
What's the reason for draft PR? :)
LGTM!

File: conf/defaults.ini (hunk 1 above)
Comment (marefr) [reply conf/defaults.ini]: @bergquist What do you mean, you want this to default to true if in dev environment?

Comment (bergquist) [reply conf/defaults.ini]: No. I just wanted to mention that I think you made the right decision to enable these headers by default :)

Comment (marefr) [reply conf/defaults.ini]: Great :+1:

**Pull request merged**